src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/summaries/temperature/CompactNodeSummary.java

```diff
@@ -68,6 +78,71 @@ public CompactNodeSummary(final String nodeId, final String hostAddress) {
         this.numOfShards = new int[TemperatureVector.Dimension.values().length];
     }
 
+    public static CompactNodeSummary buildSummaryFromDatabase(Result<Record> records,
+        DSLContext context) {
+        if (records.size() != 1) {
+            LOG.error("Expected 1 compact node summary, got {}. Summaries: {}", records.size(),
+                records);
+            throw new IllegalArgumentException(
+                "Only 1 CompactNodeSummary expected. Found: " + records.size());
+        }
+
+        Record record = records.get(0);
+        final String nodeId =
+            record.get(DSL.field(DSL.name(HotNodeSummary.SQL_SCHEMA_CONSTANTS.NODE_ID_COL_NAME),
+                String.class));
+        final String hostAddress =
+            record.get(DSL.field(DSL.name(SQL_SCHEMA_CONSTANTS.HOST_IP_ADDRESS_COL_NAME)), String.class);
+
+        CompactNodeSummary summary = new CompactNodeSummary(nodeId, hostAddress);
+
+        readAndSetTotalConsumedPerDimension(record, summary);
+        readAndSetNumShardsPerDimension(record, summary);
+        readAndSetTemperatureVector(record, summary);
+
+        return summary;
+    }
+
+    private static void readAndSetTemperatureVector(Record record, CompactNodeSummary summary) {
+        for (TemperatureVector.Dimension dimension : TemperatureVector.Dimension.values()) {
+            String normalizedMeanUsageForDimension = record
+                .get((DSL.field(DSL.name(dimension.NAME + MEAN_SUFFIX_KEY),
+                    String.class)));
+            short value = 0;
+            if (normalizedMeanUsageForDimension != null && !normalizedMeanUsageForDimension.isEmpty()) {
+                value = Short.parseShort(normalizedMeanUsageForDimension);
+            }
+            summary.setTemperatureForDimension(dimension,
+                new NormalizedValue(value));
+        }
+    }
+
+    private static void readAndSetNumShardsPerDimension(Record record, CompactNodeSummary summary) {
+        for (TemperatureVector.Dimension dimension : TemperatureVector.Dimension.values()) {
+            String numShardsForDimension = record
+                .get((DSL.field(DSL.name(dimension.NAME + NUM_SHARDS_SUFFIX_KEY),
+                    String.class)));
+            int value = 0;
+            if (numShardsForDimension != null && !numShardsForDimension.isEmpty()) {
+                value = Integer.parseInt(numShardsForDimension);
+            }
+            summary.setNumOfShards(dimension, value);
+        }
+    }
+
+    private static void readAndSetTotalConsumedPerDimension(Record record,
+        CompactNodeSummary summary) {
+        for (TemperatureVector.Dimension dimension : TemperatureVector.Dimension.values()) {
+            String totalConsumedForDimension =
+                record.get((DSL.field(DSL.name(dimension.NAME + TOTAL_SUFFIX_KEY), String.class)));
```

Comment:
can we simple cast the return type as Double.class here ?

record.get((DSL.field(DSL.name(dimension.NAME + TOTAL_SUFFIX_KEY), Double.class)));

Reply:
That was my first implementation. It failed with a class cast exception, so I ended up doing it this way.